src/lsm/cache_map_fuzz.zig

```diff
@@ -0,0 +1,369 @@
+        for (env.cache_map.cache.values, 0..) |*cache_value, i| {
+            // If the count for an index is 0, the value doesn't exist.
+            if (env.cache_map.cache.counts.get(i) == 0) {
+                continue;
+            }
+
+            const model_val = env.model.get(TestTable.key_from_value(cache_value));
+
+            assert(stdx.equal_bytes(Value, cache_value, &model_val.?.value));
+        }
+
+        // The stash can have stale values, but in that case the real value _must_ exist
+        // in the cache. It should be impossible for the stash to have a value that isn't in the
+        // model, since cache_map.remove() removes from both the cache and stash.
+        var stash_iterator_1 = env.cache_map.map_1.keyIterator();
+        while (stash_iterator_1.next()) |stash_value| {
+            // Get account from model.
+            const model_value = env.model.getPtr(TestTable.key_from_value(stash_value));
+
+            // Even if the stash has stale values, the key must still exist in the model.
+            assert(model_value != null);
+
+            const stash_value_equal = stdx.equal_bytes(Value, stash_value, &model_value.?.value);
+
+            if (!stash_value_equal) {
+                // We verified all cache entries were equal and correct above, so if it exists,
+                // it must be right.
+                const cache_value = env.cache_map.cache.get(TestTable.key_from_value(stash_value));
+                assert(cache_value != null);
+            }
+        }
+
+        var stash_iterator_2 = env.cache_map.map_2.keyIterator();
+        while (stash_iterator_2.next()) |stash_value| {
+            // Get account from model.
+            const model_value = env.model.getPtr(TestTable.key_from_value(stash_value));
+
+            // Even if the stash has stale values, the key must still exist in the model.
+            assert(model_value != null);
+
+            const stash_value_equal = stdx.equal_bytes(Value, stash_value, &model_value.?.value);
+
+            if (!stash_value_equal) {
+                // Same logic as when map_1 checks the cache above.
+                const cache_value = env.cache_map.cache.get(TestTable.key_from_value(stash_value));
+                const map_1_value = env.cache_map.map_1.get(stash_value.*);
+                assert(cache_value != null or map_1_value != null);
+            }
+        }
+
+        log.info(
+            "Verified all items in the cache, 1st stash, 2nd stash exist and match the model.",
+            .{},
+        );
+    }
+};
+
+fn random_id(random: std.rand.Random, comptime Int: type) Int {
+    // We have two opposing desires for random ids:
+    const avg_int: Int = if (random.boolean())
+        // 1. We want to cause many collisions.
+        100 * constants.lsm_growth_factor * 2048
+    else
+        // 2. We want to generate enough ids that the cache can't hold them all.
+        constants.lsm_growth_factor * 2048;
```

Comment:
Aren't these cases flipped? If we want many collisions, the average should be small, no?

Reply:
I copied the tree fuzzer to get started; it has the same comment and yeah it's wrong (forest fuzz has them correctly)!